### PR TITLE
Corestuff

### PR DESCRIPTION
**Pull request by athendrix**
I fixed our project, and it's now ready to turn in as far as I can tell.
Perhaps we should do some commenting, and maybe some cleanup with removing the functions that we no longer use and whatnot, but as far as I can tell it works, and we could be done!
:)

### CODE REVIEW

File: src/solve.c [hunk 1]
```diff
@@ -28,7 +28,30 @@
...
   LOG("solve CALLED", 2);
-  int toReturn = process_and(in);
+  int nbvars = in->nbvars;
+  //int toReturn = process_and(in);
+  //int * counts = (int*)malloc(sizeof(int)* nbvars);
+  int * setvals = (int*)malloc(sizeof(int)* nbvars);
+  int i;
+  for (i = 0; i < nbvars; i++)
+  {
+    //counts[i] = 0;
```
Comment (JaredPotter) [inline src/solve.c]: Line 38. Can this be removed?

File: src/solve.c [hunk 2]
```diff
@@ -28,7 +28,30 @@
...
+    toReturn = 1;
+  }
+  else
+  {
+    setvals[0] = !setvals[0];
+    toReturn = solver(in->data, nbvars, in->nbclauses, in->clause_lengths, in->value_sums, setvals, 0, 1);
+  }
+
+
+  //free(counts);
```
Comment (JaredPotter) [inline src/solve.c]: Line 53. Can this be removed? Also, is the solution memory leak free?

**Comment by JaredPotter** [comment]
My only general recommendations would be to add some comments throughout the code to help readability. And if you want, you could add method signatures to your helper methods.

File: src/solve.c (hunk 1 above)
Comment (athendrix) [inline src/solve.c]: Yes. That was a temporary filler piece until you finished your "value_sums"

File: src/solve.c (hunk 2 above)
Comment (athendrix) [inline src/solve.c]: Same as above, and it should be memory leak free, I just have a single malloc and a single free in the same method.

**Pull request merged**